src/Microsoft.OData.Client/Materialization/EnumValueMaterializationPolicy.cs

```diff
@@ -40,9 +40,9 @@ public object MaterializeEnumTypeProperty(Type valueType, ODataProperty property
             object materializedValue = null;
             ODataEnumValue value = property.Value as ODataEnumValue;
             this.MaterializeODataEnumValue(valueType, value.TypeName, value.Value, () => "TODO: Is this reachable?", out materializedValue);
-            if (!property.HasMaterializedValue())
+            if (!property.HasMaterializedValue(this.context))

```

Comment:
I wish the field was named `materializationContext` just like in other materialization policy classes. To disambiguate between `DataServiceContext` and `ODataMaterializationContext`

Reply:
Yeah, I do agree that does it does lead to some confusion, especially since in some places `context` refers to `DataServiceContext`. I can rename the field. I avoided out of fear of adding more "noise" to the PR.